javascript/src/methods.js

```diff
@@ -543,6 +543,43 @@ methods.addGeoJSON = function(data, layerId, group, style) {
   if (typeof(data) === "string") {
     data = JSON.parse(data);
   }
+  let popupProperty = style.popupProperty;
+  let labelProperty = style.labelProperty;
+  delete style.popupProperty;
+  delete style.labelProperty;
+  let popupOptions = style.popupOptions;
+  let labelOptions = style.labelOptions;
+  delete style.popupOptions;
+  delete style.labelOptions;
+
+  let defaultStyle = {};
+  let highlightStyle = style.highlightOptions;
+  delete style.highlightOptions;
+  if(!$.isEmptyObject(highlightStyle)) {
```

Comment:
It looks like the highlightStyle iteration, highlightFeature, and resetFeature are all copy/pasted from addLayers. Please refactor.